src/Service/Resolvers/CosmosMutationEngine.cs

```diff
@@ -68,6 +80,80 @@ private async Task<JObject> ExecuteAsync(IDictionary<string, object?> queryArgs,
             return response.Resource;
         }
 
+        /// <summary>
+        /// Authorization check on mutation fields provided in a GraphQL Mutation request.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <param name="parameters"></param>
+        /// <param name="entityName"></param>
+        /// <param name="mutationOperation"></param>
+        /// <exception cref="DataApiBuilderException"></exception>
+        public void AuthorizeMutationFields(
```

Comment:
Should we add this method AuthorizeMutationFields to IMutationEngine?

Reply:
Would you mean add a interface for this method? Please note that we will have a follow up PR to support the scenario with nested mutation, so this method will be updated and different from sql.

Reply:
Yes I meant to add this method to the already defined interface suggesting mutation engine need to implement `AuthorizeMutationFields`. This helps to contrast with QueryEngines where the @authorize from HotChocolate takes care of the AuthZ for us.

Reply:
Added an interface for this method.